patches/api/0421-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
++
++    // FireworkMeta
++    PropertyKey<MutablePropertyView> FIREWORKS = PropertyKey.of("Fireworks", MutablePropertyView.class);
++    PropertyKey<Byte> FLIGHT = PropertyKey.of("Flight", Byte.class);
++    PropertyKey<List<FireworkEffect>> EXPLOSIONS = PropertyKey.of("Explosions", TypeFactory.parameterizedClass(List.class, FireworkEffect.class));
++
++    // FireworkEffectMeta
++    PropertyKey<FireworkEffect> EXPLOSION = PropertyKey.of("Explosion", FireworkEffect.class);
++
++    // EnchantmentStorageMeta
++    PropertyKey<List<EnchantmentPair>> STORED_ENCHANTMENTS = PropertyKey.of("StoredEnchantments", TypeFactory.parameterizedClass(List.class, EnchantmentPair.class));
++
++    // CrossbowMeta
++    PropertyKey<Boolean> CHARGED = PropertyKey.of("Charged", Boolean.class);
++    PropertyKey<List<ItemStack>> CHARGED_PROJECTILES = PropertyKey.of("ChargedProjectiles", TypeFactory.parameterizedClass(List.class, ItemStack.class));
++
++    // CompassMeta
++    PropertyKey<Position> LODESTONE_POS = PropertyKey.of("LodestonePos", Position.class);
++    PropertyKey<NamespacedKey> LODESTONE_DIMENSION = PropertyKey.of("LodestoneDimension", NamespacedKey.class);
++    PropertyKey<Boolean> LODESTONE_TRACKED = PropertyKey.of("LodestoneTracked", Boolean.class);
++
++    // BundleMeta
++    PropertyKey<List<ItemStack>> ITEMS = PropertyKey.of("Items", TypeFactory.parameterizedClass(List.class, ItemStack.class));
++
++    // BookMeta
++    PropertyKey<String> TITLE = PropertyKey.of("title", String.class);
++    PropertyKey<String> TITLE_FILTERED = PropertyKey.of("filtered_title", String.class);
++    PropertyKey<String> AUTHOR = PropertyKey.of("author", String.class);
++    PropertyKey<List<Component>> PAGES = PropertyKey.of("pages", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<List<Component>> PAGES_FILTERED = PropertyKey.of("filtered_pages", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<BookMeta.Generation> GENERATION = PropertyKey.of("generation", BookMeta.Generation.class);
++    PropertyKey<Boolean> RESOLVED = PropertyKey.of("resolved", Boolean.class);
++
++    // UnsignedBookMeta
++    PropertyKey<List<String>> PAGES_WRITTEN = PropertyKey.of("pages", TypeFactory.parameterizedClass(List.class, String.class));
++
++    // BlockStateMeta
++    PropertyKey<MutablePropertyView> BLOCK_ENTITY_TAG = PropertyKey.of("BlockEntityTag", MutablePropertyView.class);
++    // Some odd entity tags included in item meta
++    PropertyKey<NamespacedKey> NOTE_BLOCK_SOUND = PropertyKey.of("note_block_sound", NamespacedKey.class);
++
++    // AxolotlBucket
++    PropertyKey<Axolotl.Variant> VARIANT = PropertyKey.of("Variant", Axolotl.Variant.class);
++
++    // MusicInstrumentMeta
++    PropertyKey<MusicInstrument> INSTRUMENT = PropertyKey.of("instrument", MusicInstrument.class);
++
++    // BannerMeta
++    PropertyKey<DyeColor> SHIELD_BANNER_BASE = PropertyKey.of("Base", DyeColor.class);
++
++}
```

Comment:
Ok, so some more missing properties are `NoAI`, `Silent`, `NoGravity`, `Glowing`, `Invulnerable`, and `Health` for the other mob bucket mobs like pufferfish, salmon, cod. 

Tadpole is also missing the above ones as well as `Age`.